File: VindsidenApp/RHCNavigationViewController.m
//
//  RHCNavigationViewController.m
//  VindsidenApp
//
//  Created by Ragnar Henriksen on 04.05.13.
//  Copyright (c) 2013 RHC. All rights reserved.
//

#import "RHCNavigationViewController.h"

@interface RHCNavigationViewController ()

@end

@implementation RHCNavigationViewController


- (NSUInteger)supportedInterfaceOrientations
{
    if ( [self.topViewController isKindOfClass:NSClassFromString(@"RHEWebCamViewController")] ) {
        return UIInterfaceOrientationMaskAllButUpsideDown;
    }
    return UIInterfaceOrientationMaskPortrait;
}


@end

File: VindsidenApp/RHCNavigationViewController.swift
//
//  RHCNavigationViewController.swift
//  VindsidenApp
//
//  Created by Ragnar Henriksen on 18/09/14.
//  Copyright (c) 2014 RHC. All rights reserved.
//

import UIKit


@objc(RHCNavigationViewController)
class RHCNavigationViewController : UINavigationController
{

    override func supportedInterfaceOrientations() -> UIInterfaceOrientationMask {
        return .Portrait
    }


    override func shouldAutorotate() -> Bool {
        return false
    }


    override func preferredInterfaceOrientationForPresentation() -> UIInterfaceOrientation {
        return .Portrait
    }
}
